### ISSUE TITLE
Subtract global mean twice in kaldi/egs/voxceleb/v2/run.sh 

### ISSUE THREAD

**Issue by zixuan2** [kaldi-asr/kaldi#3882]
kaldi/egs/voxceleb/v2/run.sh uses following as ivector-plda-scoring input:
"ark:ivector-subtract-global-mean $nnet_dir/xvectors_train/mean.vec scp:$nnet_dir/xvectors_voxceleb1_test/xvector.scp ark:- | transform-vec $nnet_dir/xvectors_train/transform.mat ark:- ark:- | ivector-normalize-length ark:- ark:- |"

Because LDA mat is an affine transform, it will subtract global mean so it seems above code subtract global mean twice. I remove $nnet_dir/xvectors_train/mean.vec and final eer drops from ~3.1% to ~2.8%.

**Comment by danpovey** [comment]
If there is length normalization in between the two normalizations that
would make it nontrivial, since length normalization is nonlinear.

**Comment by zixuan2** [comment]
Thanks for your reply, but according to my experiment the results always get better if not subtract $nnet_dir/xvectors_train/mean.vec, here are my results:
1.LDA + PLDA Scoring:
Origin:
EER: 3.118%  
minDCF(p-target=0.01): 0.3329  
minDCF(p-target=0.001): 0.5268
Modified:
EER: 2.884%  
minDCF(p-target=0.01): 0.3144  
minDCF(p-target=0.001): 0.4664  

2.LDA + Cosine Scoring:   
Origin:
EER: 5.79%  
minDCF(p-target=0.01): 0.5651  
minDCF(p-target=0.001): 0.7464 
Modified:
EER: 5.483%  
minDCF(p-target=0.01): 0.5723  
minDCF(p-target=0.001): 0.7037  

3.
Origin:
EER: 4.459%  
minDCF(p-target=0.01): 0.4811  
minDCF(p-target=0.001): 0.7583 
Modified:
EER: 4.146%  
minDCF(p-target=0.01): 0.4438  
minDCF(p-target=0.001): 0.7254

**Comment by danpovey** [comment]
I dont have a lot of experience with this but I'd worry about how large
your test set is and therefore how robust the finding is.

**Comment by zixuan2** [comment]
The test dataset is voxceleb1 test ,same with kaldi/egs/voxceleb/v2
The first result is got by runing kaldi/egs/voxceleb/v2/run.sh, and second is just using Cosing Scoring instead of PLDA, and the third using ResNet as feature extractor not TDNN with training dataset only voxceleb2.

**Comment by zixuan2** [comment]
Now I got it, I used ivector-subtract-global-mean wrongly by subtracting test data mean, and that lead to a better result.

**Comment by xuanjihe** [comment]
> Now I got it, I used ivector-subtract-global-mean wrongly by subtracting test data mean, and that lead to a better result.

In this case, by subtracting test data mean would get a better result due to the channel of the test set and training set may mismatch?
